.circleci/config.yml

```diff
@@ -76,24 +79,8 @@ jobs:
           paths:
             - ~/.cache/yarn
       - run:
-          name: Build types
-          command: yarn workspace @requestnetwork/types run build
-      - run:
-          name: Build utils
-          command: yarn workspace @requestnetwork/utils run build
-      - run:
-          name: Build currency
-          command: yarn workspace @requestnetwork/currency run build
-      - run:
-          name: Generate Smart Contract types
-          command: yarn workspace @requestnetwork/smart-contracts run build
-      - run:
-          name: Generate Payment Detection queries
-          command: yarn workspace @requestnetwork/payment-detection run codegen
-      - run:
-          name: Build all packages (tsc)
-          command: yarn build:tsc
-
+          name: Build all packages with lerna
+          command: 'yarn build --since master'
```

Comment:
As far as I understand, the CI was constructed this way to only "build" packages required for the tests to run, which optimizes the time it takes to run this job overall.

For instance, looking at the [last execution on master](https://app.circleci.com/pipelines/github/RequestNetwork/requestNetwork/10303/workflows/cc2cd86b-6599-4541-ad76-8efdd73031ff), the build step took 3m28s to run, whereas it [now takes](https://app.circleci.com/pipelines/github/RequestNetwork/requestNetwork/10346/workflows/9dbb3ac5-e795-4084-ab55-6e5dbf29c814) 4min40s to complete after this change.

Using `yarn build` runs scripts that are non required for tests, like the script that distributes the request-client code to UMD with webpack ([see here](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/request-client.js/package.json#L36)) so it's a slow process.

If I understand correctly, thanks to Nx, this build is cached and will accelerate the `next-release` job. So overall, it will speed up the CI on master. We might lose a bit of performance for PR commits, but gain some for commits on master. Looks OK to me 👍

Reply:
Exactly, one of my goals was to simplify it without loosing on performance.